### PR TITLE
Processing Data in Models

### PR DESCRIPTION
**Pull request by iamchristianani**
## Processing Data in Models for the Blog App

> In this Pull Request, I was able to do the following:

- Create models
  - Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
  - Set up associations between models.
- Made use of models to insert data
  - Created at least one user.
  - Created at least 4 posts written by one of the users.
  - Created at least 6 posts comments for one of the posts.
  - Made use of other CRUD methods to find, update and delete entities.
- Created custom methods for
  - Users
    - A method that returns the 3 most recent posts for a given user.
  - Posts
    - A method that updates the posts counter for a user.
    - A method which returns the 5 most recent comments for a given post.
  - Comments
    - A method that updates the comments counter for a post.
  - Likes
    - A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
```
Comment (franclobo) [inline app/models/user.rb]: - [ ] Please 🙏🏽 specify the foreign key for comments and likes as you have done for the post e.g. `has_many :comments, foreign_key: 'author_id'` defining foreign key helps to index the items from an external table.

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  after_create :update_posts_counter
```
Comment (franclobo) [inline app/models/post.rb]: - **[OPTIONAL]** We can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private.

```
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @iamchristianani, 

# ♻️ Required Changes ♻️

so-close-jim-halpert

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- Professional ReADME file ✔️ 
- All models created 💯 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @iamchristianani , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**